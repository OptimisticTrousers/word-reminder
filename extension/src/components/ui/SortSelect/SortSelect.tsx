import CSSModules from "react-css-modules";

import styles from "./SortSelect.module.css";

interface Props {
  disabled: boolean;
  required: boolean;
}

export const SortSelect = CSSModules(
  function ({ disabled, required }: Props) {
    return (
      <div styleName="sort">
        <label styleName="sort__label">
          Sort by:
          <select
            styleName="sort__select"
            name="created_at"
            disabled={disabled}
            required={required}
          >
            <option styleName="sort__option" value="">
              Featured
            </option>
            <option styleName="sort__option" value="1">
              Newest
            </option>
            <option styleName="sort__option" value="-1">
              Oldest
            </option>
          </select>
        </label>
      </div>
    );
  },
  styles,
  { allowMultiple: true, handleNotFoundStyleName: "log" }
);
