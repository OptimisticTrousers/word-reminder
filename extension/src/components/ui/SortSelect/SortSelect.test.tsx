import { render, screen } from "@testing-library/react";
import { SortSelect } from "./SortSelect";

describe("SortSelect", () => {
  it("renders select with three options: Featured, Newest, Oldest", async () => {
    const { asFragment } = render(
      <SortSelect disabled={false} required={false} />
    );

    const select = screen.getByLabelText("Sort by:");
    const featuredOption = screen.getByRole("option", { name: "Featured" });
    const newestOption = screen.getByRole("option", { name: "Newest" });
    const oldestOption = screen.getByRole("option", { name: "Oldest" });

    expect(select).toBeInTheDocument();
    expect(select).not.toBeDisabled();
    expect(select).not.toBeRequired();
    expect(featuredOption).toBeInTheDocument();
    expect(newestOption).toBeInTheDocument();
    expect(oldestOption).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("disables select", async () => {
    render(<SortSelect disabled={true} required={false} />);

    const select = screen.getByLabelText("Sort by:");
    const featuredOption = screen.getByRole("option", { name: "Featured" });
    const newestOption = screen.getByRole("option", { name: "Newest" });
    const oldestOption = screen.getByRole("option", { name: "Oldest" });

    expect(select).toBeInTheDocument();
    expect(select).toBeDisabled();
    expect(featuredOption).toBeInTheDocument();
    expect(newestOption).toBeInTheDocument();
    expect(oldestOption).toBeInTheDocument();
  });

  it("requires select", async () => {
    render(<SortSelect disabled={false} required={true} />);

    const select = screen.getByLabelText("Sort by:");
    const featuredOption = screen.getByRole("option", { name: "Featured" });
    const newestOption = screen.getByRole("option", { name: "Newest" });
    const oldestOption = screen.getByRole("option", { name: "Oldest" });

    expect(select).toBeInTheDocument();
    expect(select).toBeRequired();
    expect(featuredOption).toBeInTheDocument();
    expect(newestOption).toBeInTheDocument();
    expect(oldestOption).toBeInTheDocument();
  });
});
