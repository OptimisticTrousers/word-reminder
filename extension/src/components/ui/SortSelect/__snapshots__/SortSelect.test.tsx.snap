// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SortSelect > renders select with three options: Featured, Newest, Oldest 1`] = `
<DocumentFragment>
  <div
    class="_sort_7b9046"
  >
    <label
      class="_sort__label_7b9046"
    >
      Sort by:
      <select
        class="_sort__select_7b9046"
        name="created_at"
      >
        <option
          class="_sort__option_7b9046"
          value=""
        >
          Featured
        </option>
        <option
          class="_sort__option_7b9046"
          value="1"
        >
          Newest
        </option>
        <option
          class="_sort__option_7b9046"
          value="-1"
        >
          Oldest
        </option>
      </select>
    </label>
  </div>
</DocumentFragment>
`;
